### PR TITLE
lua-lpeg: Add patch to build dylib; neovim-devel: Update to 2023-06-24

### PR DESCRIPTION
**Pull request by judaew**
#### Description

Neovim now uses lua-lpeg for runtime. By default, Lpeg is compiled with the `-share` flag, which creating a bundle library. Neovim tries to link the bunlde library with the dylib library, which results in the following error:

```
:info:build ld: can't link with bundle (MH_BUNDLE) only dylibs (MH_DYLIB) file '/opt/local/lib/lua/5.1/lpeg.so' for architecture x86_64
:info:build clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

The patch adds the dylib library (with the `-dynamiclib`) in addition to the bundle library (`lpeg.so` and `lpeg.dylib`). Thus, the linker selects `lpeg.dylib` for building and Neovim builds and work correctly.

See https://github.com/neovim/neovim/issues/24124.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.4 22F66 x86_64
Xcode 15.0 15A5160n

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lua/lua-lpeg/Portfile [hunk 1]
```diff
@@ -1,23 +1,56 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           luarocks_org 1.0
```
Comment (l2dy) [inline lua/lua-lpeg/Portfile:4]: Did you test `port livecheck` after this change?

File: lua/lua-lpeg/Portfile [hunk 2]
```diff
@@ -1,23 +1,56 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
```
Comment (l2dy) [inline lua/lua-lpeg/Portfile:1]: Please resolve the lint error `Error: Missing required variable: categories`.

**Review by l2dy — CHANGES_REQUESTED** [review]
In general, we should keep using the `luarocks_org` PortGroup if possible.

**Comment by judaew** [comment]
> In general, we should keep using the luarocks_org PortGroup if possible.

I understand this, but unfortunately Neovim (~v0.10) will requires a library in dylib format, for the linker. This requires modifying the Makefile. Otherwise, there will be the following error:

```
ld: can't link with bundle (MH_BUNDLE) only dylibs (MH_DYLIB) file
```

**Comment by l2dy** [comment]
Using a PortGroup does not conflict with having a patch. I think you could still patch sources after the extract phase (i.e. `luarocks unpack` as set up in `luarocks_org`).

**Comment by judaew** [comment]
My apologies, I thought that LuaRocks is a binary repository. I'll fix it now.

**Comment by judaew** [comment]
Done

**Review by l2dy — APPROVED** [review]

**Comment by judaew** [comment]
@andremm Let me know if you are the maintainer of `lua-lpeg` port.

**Comment by andremm** [comment]
> @andremm Let me know if you are the maintainer of lua-lpeg port.

I'm sorry @judaew , but I'm not the maintainer anymore.

**Comment by judaew** [comment]
All right, then I'll mark myself as a maintainer and delete you. Thanks for supporting this port in the past!

**Comment by andremm** [comment]
Thanks for keeping it updated!

**Pull request merged**